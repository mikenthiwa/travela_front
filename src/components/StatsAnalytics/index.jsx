import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const StatsAnalytics = ({color, stats, icon, error }) => (
  <Fragment>
    { error 
      ? (
        <p className="dashboard-component__error-text--style">
          Oops! An error occurred in retrieving this data
        </p>
      )
      : (
        <div className={`analytics-card__stats ${color}`}>
          <div className="analytics-card__icon">
            <img src={icon} alt="" />
          </div>
          <div className="analytics-card__stat">
            <h4>{stats}</h4>
          </div>
        </div>
      )
    }
  </Fragment>
);

StatsAnalytics.propTypes = {
  color: PropTypes.string,
  stats: PropTypes.number,
  icon: PropTypes.string,
  error: PropTypes.string,
};

StatsAnalytics.defaultProps = {
  color: '',
  icon: '',
  stats: 0,
  error: '',
};

export default StatsAnalytics;
