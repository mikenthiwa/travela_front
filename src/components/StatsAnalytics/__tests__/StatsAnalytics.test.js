import React from 'react';
import { shallow } from 'enzyme';
import StatsAnalytics from '..';

const props = {
  color: '#000',
  stats: 40,
  icon: 'sfdhd.d'
};

describe('<Dashboard />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<StatsAnalytics {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an error text on error with loading data', () => {
    const error = 'Oops! An error occurred in retrieving this data';
    const wrapper = shallow(<StatsAnalytics {...props} error={error} />);
    expect(wrapper.find('p').text()).toEqual(error);
  });
});
