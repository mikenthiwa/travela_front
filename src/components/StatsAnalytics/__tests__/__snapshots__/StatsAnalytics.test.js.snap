// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<Fragment>
  <div
    className="analytics-card__stats #000"
  >
    <div
      className="analytics-card__icon"
    >
      <img
        alt=""
        src="sfdhd.d"
      />
    </div>
    <div
      className="analytics-card__stat"
    >
      <h4>
        40
      </h4>
    </div>
  </div>
</Fragment>
`;
